### ISSUE TITLE
externals appear as managed & unmanaged

### ISSUE THREAD

**Issue by fboundp** [twpayne/chezmoi#2778]
## Describe the bug

I use the `unmanaged` subcommand to identify new items that I might want to `chezmoi add`. However, externals appear in `unmanaged` output … and they appear in `managed` output as well.

## To reproduce

- Add an external, e.g. in `$XDG_CONFIG_HOME/chezmoi/.chezmoiexternal.toml`:

```
[".zsh/zsh-z"]
    url = "https://github.com/agkozak/zsh-z.git"
    type = "git-repo"
    refreshPeriod = "168h"
```

- `chezmoi -R apply`
- ✅ Run `chezmoi managed` to confirm it is managed:

```console
% chezmoi managed | grep zsh-z
.zsh/zsh-z
```

- ❌️ Run `chezmoi unmanaged` to confirm there is nothing unexpectedly unmanaged:

```console
% chezmoi unmanaged | grep zsh-z
.zsh/zsh-z/.git
.zsh/zsh-z/.github
.zsh/zsh-z/.gitignore
.zsh/zsh-z/LICENSE
.zsh/zsh-z/README.md
.zsh/zsh-z/_zshz
.zsh/zsh-z/img
.zsh/zsh-z/zsh-z.plugin.zsh
```

## Expected behavior

`chezmoi unmanaged` would not list externals.

**Comment by halostatue** [comment]
This looks like it should be changed or *able* to be changed, but I’m not quite sure whether this is a "bug" or a "feature", because I can see arguments either way. Strictly speaking, externals *are not managed* by chezmoi, but they are *placed* by chezmoi.

If externals should be excluded by default from `chezmoi unmanaged`, then this is a bug. If they should be excluded on  request (`chezmoi unmanaged --exclude-externals`), then it’s a feature request.

**Comment by fboundp** [comment]
I agree and I’m comfortable with the ambiguity. In fact `chezmoi managed --exclude=externals` is a thing, and so I thought `chezmoi unmanaged --exclude=externals` might also be a thing. (It’s not.)

**Comment by halostatue** [comment]
Based on that, I think that extending `chezmoi unmanaged --exclude=externals` should be a thing because of the ambiguity.

**Comment by twpayne** [comment]
The issue here is that the external is of type `git-repo`. This means that chezmoi is limited to running `git init` and `git pull` in the directory, and chezmoi has no insight into the contents of the directory.

In effect, the directory is managed by chezmoi but its contents are not, which is why `.zsh/zsh-z` appears in the output of `chezmoi managed` but not in the output of `chezmoi unmanaged`. Vice versa for the contents of `.zsh/zsh-z`: the contents appear in `chezmoi unmanaged` but the directory `.zsh/zsh-z` does not.

If you use an external of type `archive` instead of `git-repo`, e.g.

```toml
[".zsh/zsh-z"]
    url = "https://github.com/agkozak/zsh-z/archive/master.tar.gz"
    type = "archive"
    exact = true
    stripComponents = 1
    refreshPeriod = "168h"
```

then chezmoi will manage both the directory and its contents, and `chezmoi managed` and `chezmoi unmanaged` will behave as you expect.

I'll update the documentation to include this.

**Comment by twpayne** [comment]
For info, after discussion in #2858, I'll change the behavior of `chezmoi unmanaged` here. See #2862.
